content/resources/how-to-guides/late-arriving-data.md

```diff
@@ -0,0 +1,178 @@
+
+`stop = date.truncate(t: now(), unit: every. `The `late_window` is equal to the longest amount of time you're willing to wait for late arriving data (in this example it’s equal to ). The date.truncate() function is used to truncate the start and stop time to the latest minute to ensure that you successfully recompute values on the same timestamps.  Where `every = task.every`. Since the task is running at 1 minute intervals `every `is equal to `1m`. Additionally remember that the aggregateWindow function uses the _stop column as the source of the new time value for aggregate values by default. 
+
+Next, compute the current mean and count with the following code: \
+`// Always compute the most recent interval`
+
+
+```
+newCounts
+    |> filter(fn: (r) => r._time == stop)
+    |> map(
+        fn: (r) => {
+            response = invokeScript(start: date.sub(d: every, from: r._time), stop: r._time)
+
+            return {r with code: response.statusCode}
+        },
+    )
+    |> yield(name: "current")
+```
+
+
+filter for the last `newCount` value. Remember, the time value is equal to the stop value because of the default behavior of the aggregateWindow() function. Then map over that single row table to call the invokeScript function once. Here you also pass in values to the start and stop parameters of `date.sub(d: every, from: r._time)` and `r._time`, respectively.  Remember that the `every` variable is equal to 1m. Effectively, this means that you will calculate the mean and count over 1minute intervals (with timestamps appropriately truncated to ensure overwrites of recomputed means later). This code ensures that you will always invoke the water_level_process.flux script at least once to write new means and counts to the water_level_mean and water_level_checksum buckets, respectively. 
+
+Next, query the water_level_checksum bucket for the last hour: 
+
+
+```
+oldCounts =
+    from(bucket: "water_level_checksum")
+        |> range(start: start, stop: stop)
+        |> group(columns: ["_measurement", "_field"])
+```
+
+
+Remember the start and stop times here are equal to - and now() truncated to the minute. 
+
+Now join the old counts and new counts together. You also filter for when the counts differ. If they do differ, then there will be records in the response that can be mapped over. Map over those records to recalculate the mean and count by invoking the level_water_process.flux script: 
+
+
+```
+experimental.join(
+    left: oldCounts,
+    right: newCounts,
+    fn: (left, right) => ({left with old_count: left._value, new_count: right._value}),
+)
+    // Recompute any windows where the checksum is different
+    |> filter(fn: (r) => r.old_count != r.new_count)
+    |> map(
+        fn: (r) => {
+            response = invokeScript(start: date.sub(d: every, from: r._time), stop: r._time)
+
+            return {r with code: response.statusCode}
+        },
+    )
+    |> yield(name: "diffs")
+```
```

Comment:
Explain the task in the proposed "Task details" section, but show the task in it's entirety.

Reply:
Thank you! All comments have been addressed.